### PR TITLE
fix(calendar): vertical appearance in mobile devices

### LINKED ISSUE #3173: It is not possible to change the calendar display on a mobile
### NextUI Version

2.4.0

### Describe the bug

It is not possible to change the calendar display on a mobile phone. it should be possible to display one by one as a column

Снимок экрана 2024-06-04 в 13 49 57
image

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1) add DateRangePicker with 2 visibleMonths to the page
2) switch to mobile mode
3) open calndar

### Expected behavior

calendars displayed one by one as column, no scrollbars
Снимок экрана 2024-06-04 в 13 55 17

### Screenshots or Videos

_No response_

### Operating System Version

macOs

### Browser

Chrome

### PR DESCRIPTION
**Pull request by awesome-pro**
Closes #3173
## 📝 Description

 Added the logic for the vertical appearance of calendars (if more than one) in case of mobile devices

## ⛳️ Current behavior (updates)

https://github.com/user-attachments/assets/7ce12229-a825-4973-a114-9652581219f0

https://github.com/user-attachments/assets/768a234e-dde9-4edc-998d-09e06f96d8ea

> Date Picker

https://github.com/user-attachments/assets/987f67d6-16af-4037-b3f6-e2426bd1d7fa

## Summary by CodeRabbit

- **New Features**
	- Enhanced responsiveness of the calendar component for mobile devices.
	- Added a new property to specify the calendar height, improving configurability for both calendar and date picker components.

- **Bug Fixes**
	- Fixed navbar height issues for better visual consistency.

- **Style Improvements**
	- Updated CSS styles for the calendar and date picker components to enhance layout and responsiveness across devices.

### CODE REVIEW

**Comment by wingkwong** [comment]
@awesome-pro because you're using pnpm@v9 which is not supported currently. I updated it back to v8 at this moment. We plan to upgrade to v9 and I made a PR for that. It can be easily to obverse if you look at the pnpm-lock.yaml. I'll review the rest when I got time.

**Comment by awesome-pro** [comment]
Thanks @wingkwong

File: packages/components/calendar/src/calendar-base.tsx [hunk 1]
```diff
@@ -93,13 +96,42 @@ export function CalendarBase(props: CalendarBaseProps) {
...
+      <div className="flex flex-col gap-0">
+        {isMobile && (
+          <Fragment key={`calendar-header-${i}`}>
+            {i === 0 && (
+              <div className="w-full flex items-center justify-between">
+                <Button
+                  {...prevButtonProps}
+                  onPress={chain(prevButtonProps.onPress, () => setDirection(-1))}
+                >
+                  {rtlDirection === "rtl" ? <ChevronRightIcon /> : <ChevronLeftIcon />}
```
Comment (wingkwong) [inline packages/components/calendar/src/calendar-base.tsx:108]: in RTL, we don't switch left and right arrow icon.

File: packages/components/calendar/src/calendar-base.tsx [hunk 2]
```diff
@@ -93,13 +96,42 @@ export function CalendarBase(props: CalendarBaseProps) {
...
+                </Button>
+
+                <Button
+                  {...nextButtonProps}
+                  onPress={chain(nextButtonProps.onPress, () => setDirection(1))}
+                >
+                  {rtlDirection === "rtl" ? <ChevronLeftIcon /> : <ChevronRightIcon />}
```
Comment (wingkwong) [inline packages/components/calendar/src/calendar-base.tsx:115]: in RTL, we don't switch left and right arrow icon.

File: packages/components/calendar/src/use-calendar-base.ts [hunk 3]
```diff
@@ -48,6 +48,13 @@ interface Props extends NextUIBaseProps {
    * @default 256
    */
   calendarWidth?: number | string;
+  /**
+   * The height to be applied to the calendar component. This value is multiplied by the number
+   * of visible months to determine the total height of the calendar.
+   *
+   * @default 274
+   */
+  calendarHeight?: number | string;
```
Comment (wingkwong) [inline packages/components/calendar/src/use-calendar-base.ts:57]: update docs as well

**Review by wingkwong — CHANGES_REQUESTED** [review]
The style is a bit different, e.g the space between the month and year and the calendar

image

**Comment by wingkwong** [comment]
Closing due to inactivity

**Pull request closed without merging**